src/source/amplify.rs

```diff
@@ -53,7 +97,9 @@ where
 
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
-        self.input.next().map(|value| value * self.factor)
+        self.input
+            .next()
+            .map(|value| value * f32::from(self.factor.clone()))
```

Comment:
this is running `f32::from()` and `factor.clone()` about 96 thousand times a second. Instead you could keep this as a simple multiply and do the `f32::from()` in `set_factor`.